feat: resolves issue 33572 re aria-sort on tables

dgreene1 opened the pull request:
<!--
First of all, thank you for your contribution! 😄

New feature please send a pull request to feature branch, and rest to master branch.
Pull requests will be merged after one of the collaborators approve.
Please makes sure that these forms are filled before submitting your pull request, thank you!
-->

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

<!--
1. Describe the source of requirement, like related issue link.
-->
closes https://github.com/ant-design/ant-design/issues/33572

### 💡 Background and solution

<!--
1. Describe the problem and the scenario.
2. GIF or snapshot should be provided if includes UI/interactive modification.
3. How to fix the problem, and list final API implementation and usage sample if that is a new feature.
-->

The problem was described in https://github.com/ant-design/ant-design/issues/33572.

No API changes were made.

The only difference is that the `aria-sort` attribute shows up now on the header tag. For example: `aria-sort="descending"`

### 📝 Changelog

<!--
Describe changes from the user side, and list all potential break changes or other risks.
--->

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Table adds `aria-sort` attribute for screen readers.  |
| 🇨🇳 Chinese | Table 增加 aria-sort 属性以优化屏幕阅读器的使用体验。 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [X] Doc is updated/provided or **not needed**
- [X] **Demo is updated/provided** or not needed
- [X] TypeScript definition is updated/provided or **not needed**
- [X] **Changelog is provided** or not needed

dgreene1 committed:
feat: resolves issue 33572 re aria-sort on tables

afc163 commented:
<img width="876" alt="图片" src="https://user-images.githubusercontent.com/507615/148497305-cf6b0f6a-fec4-4ec9-9116-1f7415004e23.png">

dgreene1 committed:
fixing snapshots

afc163 commented:
<img width="569" alt="图片" src="https://user-images.githubusercontent.com/507615/148711692-af3d6161-6dd8-4c07-b4a5-4e94d1737e0b.png">

coverage percent is down.

dgreene1 commented:
@afc163 I’m surprised that coverage went down since I added unit tests and enhanced many other tests. I will bring it back to 100% tomorrow (Monday).

dgreene1 committed:
removing code that can't be hit anyway

dgreene1 commented:
@afc163 I removed the one line that isn't actually reachable by tests. I had put it in there as an attempt at an "assert never" type of approach. But I think give that there is only 3 sort states (up, down, and inactive) it's fine to remove the line that I had added.

useSorter should be at 100% code coverage now.
